app_dart/lib/src/service/config.dart

```diff
@@ -326,10 +326,15 @@ class Config {
 
   Future<String> generateGithubToken(gh.RepositorySlug slug) async {
     // GitHub's secondary rate limits are run into very frequently when making auth tokens.
+    const RetryOptions retryOptions = RetryOptions(maxAttempts: 3, delayFactor: Duration(seconds: 3));
     final Uint8List? cacheValue = await _cache.getOrCreate(
       configCacheName,
       'githubToken-${slug.fullName}',
-      createFn: () => _generateGithubToken(slug),
+      // If generating the github token fails, wait 3 seconds and retry up to 3 times.
+      createFn: () => retryOptions.retry(
+        () async => await _generateGithubToken(slug),
+        retryIf: (Exception e) => true,
+      ),
```

Comment:
Can we add a test? I suspect we're going to need to tweak this because the secondary rate limit is over a period of a minute.